### ISSUE TITLE
Instant Search: ensure search results matches have good contrast/legibility

### ISSUE THREAD

**Issue by keoshi** [Automattic/jetpack#14666]
When you combine our many customization options for the Instant Search overlay with themes, one can end up with completely illegible matched portions on highlights/`mark` tag.

Edit: This is _also_ valid for the close button.

See this example when using Storefront theme + JPS Light color theme + custom highlight color:

![image](https://user-images.githubusercontent.com/390760/74441976-97e2a600-4e68-11ea-80d3-27640d17904b.png)

Or this extreme example when using Storefront theme + JPS Dark color theme + custom highlight color:

![image](https://user-images.githubusercontent.com/390760/74442037-b34db100-4e68-11ea-9094-1c3785368e39.png)

The first thing I thought was to have the text color change to white or black according to the contrast ratio with the background. I [found this function online](https://gomakethings.com/dynamically-changing-the-text-color-based-on-background-color-contrast-with-vanilla-js/) that accomplishes this:

```
function getContrastYIQ(hexcolor){
	var r = parseInt(hexcolor.substr(0,2),16);
	var g = parseInt(hexcolor.substr(2,2),16);
	var b = parseInt(hexcolor.substr(4,2),16);
	var yiq = ((r*299)+(g*587)+(b*114))/1000;
	return (yiq >= 128) ? 'black' : 'white';
}
```

- Do you think it's worth pursuing this solution?
- Should we instead offer a pre-defined set of highlight colors in the Customizer that we can fully control (both background and foreground)?

**Comment by chriscoyier** [comment]
I noticed this too while testing. The default is a little too intense for Dark Mode.

<img width="1095" alt="Screen Shot 2020-03-17 at 6 23 08 AM" src="https://user-images.githubusercontent.com/69156/76862990-1d68c580-681c-11ea-9c48-603a7775859d.png">

And darkening it loses the text entirely:

<img width="1049" alt="Screen Shot 2020-03-17 at 6 24 02 AM" src="https://user-images.githubusercontent.com/69156/76863019-25c10080-681c-11ea-8598-a95bdd24ae87.png">

A programmatic solution to make the text either white or black depending on the background is definitely possible and that's not a bad choice. An alternative would be offering a second color picker to pick the color of the text itself, which opens up more design possibility which might be nice too. Or you could avoid the problem just doing underlines or something.

**Comment by keoshi** [comment]
Thanks for your feedback, @chriscoyier — truly appreciate you spending the time testing this and giving suggestions to make it better.

I believe this was shipped after the revision you're currently testing, but we now have light/dark combinations for the highlighted portion of the search query.

#### Light theme:

![image](https://user-images.githubusercontent.com/390760/76863987-4313c000-6858-11ea-89b7-a4d75639b318.png)

#### Dark theme:

![image](https://user-images.githubusercontent.com/390760/76863946-342d0d80-6858-11ea-825b-c788fb3fa172.png)

This still doesn't solve the problem that the option in the Customizer overrides these values if changed.

> An alternative would be offering a second color picker to pick the color of the text itself, which opens up more design possibility which might be nice too.

I also like this suggestion as it avoids any automated solution while offering even more flexibility (at the cost of too much flexibility perhaps?).

We'll continue to iterate on this until launch, Chris. Thanks so much once again!

**Comment by jsnmoon** [comment]
I was under the impression this was a nice-to-have instead of a must-have for the upcoming release. I've placed this issue back into the backlog. 

cc @gibrown @keoshi

**Comment by gibrown** [comment]
I'm fine with that. I wasn't sure because I thought it came up from css-tricks.
